src/lexer.cpp

```diff
@@ -264,6 +264,7 @@ Token Lexer::build_next_token() {
         case Token::Type::LParen:
         case Token::Type::Match:
         case Token::Type::Newline:
+        case Token::Type::WhenKeyword:
```

Comment:
I think we might want this same behavior for _all_ keywords, yes? If so, we can use an `if (m_last_token.is_keyword()) ...`.

Reply:
@seven1m I don't think that'll work, for a lot of keywords MRI throws a syntax error, and the def keyword is already special cased for `def /` definitions. Looks like it also needs to be supported for if/elsif/unless, while/until, rescue, and return. MRI parses those (and emits a warning message for the conditionals). How do you want to structure it? Whitelist those keywords with a `can_precede_regexp_literal` or similar method on Token?

Reply:
OK, very good, that makes sense.

>How do you want to structure it? Whitelist those keywords with a `can_precede_regexp_literal` or similar method on Token?

I like that. <3

Reply:
The lexer change is straightforward but there's an additional complication with some of those other keywords—with RubyParser they return the regexp wrapped in a `:match` node which isn't handled by NatalieParser. Working through it... couple more keywords still to do.

Reply:
OK, no worries. If you just want to ship this one as-is and come back to the other ones, that is fine too!

Reply:
Closing in favour of #12